qa/rpc-tests/test_framework/test_framework.py

```diff
@@ -172,12 +173,15 @@ def main(self):
                 # Dump the end of the debug logs, to aid in debugging rare
                 # travis failures.
                 import glob
-                filenames = glob.glob(self.options.tmpdir + "/node*/regtest/debug.log")
+                filenames = [self.options.tmpdir + "/test_framework.log"]
+                filenames += glob.glob(self.options.tmpdir + "/node*/regtest/debug.log")
                 MAX_LINES_TO_PRINT = 1000
                 for f in filenames:
-                    print("From" , f, ":")
-                    from collections import deque
-                    print("".join(deque(open(f), MAX_LINES_TO_PRINT)))
+                    try:
+                        print("From" , f, ":")
+                        print("".join(deque(open(f), MAX_LINES_TO_PRINT)))
+                    except:
```

Comment:
Agree on catching only a more concrete exception, specially if you're expecting one in particular, as it seems to be the case in https://github.com/bitcoin/bitcoin/pull/9780/commits/a2a565180449b80c054c05c12c93097f1e8752c2 . Also you could print the message of the exception too (before the traceback).